src/clib/lib/enkf/field.cpp

```diff
@@ -115,65 +114,20 @@ void field_export3D(const field_type *field, void *_target_data,
 
         field_fload_keep_inactive(initial_field, init_file);
     }
-    switch (ecl_type_get_type(data_type)) {
-    case (ECL_DOUBLE_TYPE): {
-        const double *src_data = (const double *)field->data;
-        const double *initial_src_data =
-            initial_field ? (const double *)initial_field->data : NULL;
-
-        if (ecl_type_is_float(target_data_type)) {
-            float *target_data = (float *)_target_data;
-            EXPORT_MACRO;
-        } else if (ecl_type_is_double(target_data_type)) {
-            double *target_data = (double *)_target_data;
-            EXPORT_MACRO;
-        } else {
-            fprintf(stderr,
-                    "%s: double field can only export to double/float\n",
-                    __func__);
-            abort();
-        }
-    } break;
-    case (ECL_FLOAT_TYPE): {
-        const float *src_data = (const float *)field->data;
-        const float *initial_src_data =
-            initial_field ? (const float *)initial_field->data : NULL;
-        if (ecl_type_is_float(target_data_type)) {
-            float *target_data = (float *)_target_data;
-            EXPORT_MACRO;
-        } else if (ecl_type_is_double(target_data_type)) {
-            double *target_data = (double *)_target_data;
-            EXPORT_MACRO;
-        } else {
-            fprintf(stderr, "%s: float field can only export to double/float\n",
-                    __func__);
-            abort();
-        }
-    } break;
-    case (ECL_INT_TYPE): {
-        const int *src_data = (const int *)field->data;
-        const int *initial_src_data =
-            initial_field ? (const int *)initial_field->data : NULL;
-        if (ecl_type_is_float(target_data_type)) {
-            float *target_data = (float *)_target_data;
-            EXPORT_MACRO;
-        } else if (ecl_type_is_double(target_data_type)) {
-            double *target_data = (double *)_target_data;
-            EXPORT_MACRO;
-        } else if (ecl_type_is_int(target_data_type)) {
-            int *target_data = (int *)_target_data;
-            EXPORT_MACRO;
-        } else {
-            fprintf(stderr,
-                    "%s: int field can only export to int/double/float\n",
-                    __func__);
-            abort();
-        }
-    } break;
-    default:
-        fprintf(stderr, "%s: Sorry field has unexportable type ... \n",
+
+    const float *src_data = (const float *)field->data;
+    const float *initial_src_data =
+        initial_field ? (const float *)initial_field->data : NULL;
+    if (ecl_type_is_float(target_data_type)) {
+        float *target_data = (float *)_target_data;
+        EXPORT_MACRO;
+    } else if (ecl_type_is_double(target_data_type)) {
```

Comment:
is this not always float?

Reply:
Field data is always `float`. The target data may not be. It's determined by RMS I think.